BEPs/BEP-346.md

```diff
@@ -0,0 +1,121 @@
+  string operator = 1 [(cosmos_proto.scalar) = "cosmos.AddressString"];
+  // bucket_name defines the name of the bucket where the object is stored.
+  string bucket_name = 2;
+  // object_name defines the name of object
+  string object_name = 3;
+  // payload_size defines size of the object's payload
+  uint64 payload_size = 4;
+  // content_type defines a standard MIME type describing the format of the object.
+  string content_type = 5;
+  // primary_sp_approval defines the approval info of the primary SP which indicates that primary sp confirm the user's request.
+  common.Approval primary_sp_approval = 6;
+  // expect_checksums defines a list of hashes which was generate by redundancy algorithm.
+  repeated bytes expect_checksums = 7;
+}
+```
+
+## 4.3 Workflow
+
+The steps to update an object are illustrated below. To users, it is similar when compared to the current create-upload object process.
+
+![](./assets/BEP-346/workflow.png)
+
+1. Users obtain approval from the primary SP where the object is stored and send the transaction to Greenfield with the 
+   message MsgUpdateObjectContent. This stores the updated object metadata via shadowObjectInfo and flags the object as 
+   updating without changing the objectInfo on-chain.
+
+2. Users upload the updated object to the primary SP. The SP will validate if the object can be updated and authenticate 
+   the user's account permission.
+
+3. The SP should keep the original and updated object data simultaneously, version in ObjectInfo or ShadowObjectInfo is 
+   also used to identify the object data persistence to ensure there is no conflict. Users should be able to `GET` 
+   the original object data at this moment as the version field has not changed.
+
+4. The primary SP sends the sealing object transaction to the Greenfield blockchain, changes to objectInfo will take effect according 
+   to meta stored in shadowObjectInfo with version increment. Users will be able to GET the updated object data.
+   In the case where SP fails to seal the object after retries, it would send tx RejectSealobject to Greenfield to reject 
+   the users request. The metadata shadowObjectInfo will be removed from chain, updated object data will be removed from SP.
+
+## 4.4 Fees & User Charges
+The current policy regarding user charges is as follows: When users create and upload an object, a minimum of 6 months 
+storage fee is reserved on Greenfield. This fee is then paid to the Storage Providers' accounts through stream payment. 
+If objects are stored for more than 6 months, the storage fee will be calculated based on the actual time.
+This rule applies to updating objects as well. When users update an object, Greenfield will deduct the payment for the 
+original object from users’s account and reserve at least 6 months of storage fee for the updated object, taking into 
+account the new object size. It's important for users to be aware that each time an object is updated, there will be 
```

Comment:
It is not clear here how the charges are calculated when the file size decreases, and how the prepaid fees are compensated.

Reply:
added